### DISCUSSION TITLE
trusty-l4t compilation fails on kirkstone branch

### DISCUSSION THREAD

**Discussion by f-gia** [OE4T/meta-tegra#1060] [Q&A] [answered]
Hi, we are trying to build the `trusty-l4t` sources in order to use the "secure samples" described [here](https://docs.nvidia.com/jetson/archives/l4t-archived/l4t-3271/index.html#page/Tegra%20Linux%20Driver%20Package%20Development%20Guide/trusty.html#wwpID0E0WE0HA). We are on branch kirkstone (using L4T 32.7.2). We get the following compilation error:  
```
| /build/tmp-glibc/work/armv8a_tegra194-oe-linux/trusty-l4t/32.7.2-r0/recipe-sysroot-native/usr/bin/aarch64-oe-linux/aarch64-oe-linux-ld: /build/tmp-glibc/work/armv8a_tegra194-oe-linux/trusty-l4t/32.7.2-r0/recipe-sysroot/usr/lib/aarch64-oe-linux/11.3.0/libgcc.a(lse-init.o): in function `init_have_lse_atomics':
| /usr/src/debug/libgcc/11.3.0-r0/build/libgcc/../../../../../../work-shared/gcc-11.3.0-r0/gcc-11.3.0/libgcc/config/aarch64/lse-init.c:44: undefined reference to `__getauxval'
| /usr/src/debug/libgcc/11.3.0-r0/build/libgcc/../../../../../../work-shared/gcc-11.3.0-r0/gcc-11.3.0/libgcc/config/aarch64/lse-init.c:44:(.text.startup+0x10): relocation truncated to fit: R_AARCH64_CALL26 against undefined symbol `__getauxval'
| make[2]: *** [make/build.mk:23: /build/tmp-glibc/work/armv8a_tegra194-oe-linux/trusty-l4t/32.7.2-r0/build/build-t186/lk.elf] Error 1

```
Any idea why?

**Comment by madisongh** [comment, marked as answer]
Looks like you're using a compiler other than the `gcc-linaro-baremetal-arm` compiler?  I'm not sure how well that will work, but try adding `-mno-outline-atomics` to the compilation flags to address that specific error you're seeing.  You may need to do that in the makefiles somewhere.

**Reply by f-gia** [reply]
We managed to compile the `hwkey-app` on our host machine using the SDK. However, we had to patch the definition of `CC` in the makefile in `Linux_for_Tegra/source/public/trusty_src/trusty/app/nvidia-sample/hwkey-agent/CA_sample/` not to override the environment variables, otherwise it would use the wrong compiler (as you pointed out).
